### PR TITLE
issue1100

### PR DESCRIPTION
**Pull request by lc-thomasberger**
Update blockView to add css class if element is onScreen.

Block config:
```
{
        "_id": "b-05",
        "_parentId": "a-05",
        "_type": "block",
        "_classes": "animated",
        "title": "b-05",
        "displayTitle": "",
        "body": "",
        "_trackingId": 0,
        "_onScreen": {
            "_isEnabled": true,
            "_classes": "wobble",
            "_percentInviewVertical": 50  // defaults to 33%
        }
    }
```
Disabled by default.
The block config shows example using animate.css animtions.

### CODE REVIEW

File: src/core/js/views/adaptView.js [hunk 1]
```diff
@@ -38,6 +39,25 @@ define([
             return this;
         },
 
+        setupOnScreenHandler: function() {
+            var onscreen = this.model.get('_onScreen');
+
+            if (onscreen && onscreen._isEnabled) {
```
Comment (tomgreenfield) [inline src/core/js/views/adaptView.js]: Would you mind terribly reversing the logic here so we can sink the indentation?
```js
if (!onscreen || !onscreen._isEnabled) return;
```

File: src/core/js/views/adaptView.js [hunk 2]
```diff
@@ -38,6 +39,24 @@ define([
             return this;
         },
 
+        setupOnScreenHandler: function() {
+            var onscreen = this.model.get('_onScreen');
+
+            if (!onscreen || !onscreen._isEnabled) return;
+
+            this.$el.on('onscreen', _.bind(function (e, m) {
```
Comment (oliverfoster) [inline src/core/js/views/adaptView.js]: can you change this  to ```this.$el('onscreen.adaptView'```?

File: src/core/js/views/adaptView.js [hunk 3]
```diff
@@ -38,6 +39,24 @@ define([
...
+
+                if (!m.onscreen) return;
+
+                var minVerticalInview = onscreen._percentInviewVertical || 33;
+
+                if (m.percentInviewVertical < minVerticalInview) return;
+
+                this.$el.addClass( onscreen._classes || 'onscreen' ).off('onscreen');
```
Comment (oliverfoster) [inline src/core/js/views/adaptView.js]: can you change this  to  ```.off('onscreen.adaptView');```?

File: src/core/js/views/adaptView.js [hunk 4]
```diff
@@ -93,6 +112,7 @@ define([
         },
 
         remove: function() {
+            this.$el.off('onscreen');
```
Comment (oliverfoster) [inline src/core/js/views/adaptView.js]: can you change this  to ```.off('onscreen.adaptView')```?

**Comment by oliverfoster** [comment]
i'm worried the ```off("onscreen")``` will break anything else that is listening, i suggest we namespace the event with ".adaptView"

**Pull request merged**